### ISSUE TITLE
The tracked metric is too low

### ISSUE THREAD

**Issue by yanghaibin-cool** [HanGuangXin/ByteTrack_ReID#10]
First of all thank you for adding Reid to bytetrack and making him open source，however,there are a few confusions when I reproduce your experimental results.
For now,the results are trained on half of MOT17 and tested on the other half of MOT17.
I run the following command：
python3 tools/train.py -f exps/example/mot/yolox_x_mot17_half.py -d 1 -b 2 --fp16 -o -c pretrained/yolox_x.pth.tar
This is the result after 80 rounds of training：
Bytetrack-Reid:
![image](https://user-images.githubusercontent.com/72022193/162562923-e313a45e-f666-42f7-8f1d-0f0a337c912b.png)
Here I also compared the original bytetrack in the mot17_train_half training, and found that the result of Bytetrack_Reid is not as high as the original, may I ask why this is :
Bytetrack:
![image](https://user-images.githubusercontent.com/72022193/162563063-30b4a4e8-164e-4a26-9f82-135e7f036ba0.png)
Then I used these two weight files to evaluate on MOT17_val_half and got the following results：
Bytrack-Reid:
![image](https://user-images.githubusercontent.com/72022193/162563133-2c08bf8f-cd33-4476-8671-e19a91fd06e5.png)
Bytetrack:
![image](https://user-images.githubusercontent.com/72022193/162563151-dea1d936-3d90-4a50-9e76-873544fe398e.png)
I did not get the same experimental results as you, may I ask where did I go wrong.
Looking forward to your reply.

**Comment by HanGuangXin** [comment]
@yanghaibin-cool Hi, there!
Your results of ByteTrack_ReID is almost the same as mine. But your ByteTrack results are much higher than mine.

I will give it a look, which may need some time?

**Comment by HanGuangXin** [comment]
@yanghaibin-cool And these is something not right in `yolox_s_mot17_half.py`. The training should use `train_half.json`, not `train.json`.

And my results (mAP 0.556 MOTA 72.6) is from yolox_s, not yolox_x. The results of origin bytetrack is mAP 0.535 MOTA 71.5.

Please tell me which model is used (s or x).

**Comment by HanGuangXin** [comment]
For another thing, it is reasonable to see performance degradation after using an additional ReID head. You can go to CSTrack for the detailed reason.

**Comment by HanGuangXin** [comment]
And from my results, adding a reid head will **not** obviously increase forward time. There are several conflicts between your results and mine.

So, looking forward to your reply!

**Comment by yanghaibin-cool** [comment]
> 

First of all, glad to hear from you.
The configuration of my computer is 2080ti, 11g video memory.
Below is the config file where I run the program:
Bytetrack-Reid:
![image](https://user-images.githubusercontent.com/72022193/162617841-fd9ab9eb-607c-45b7-8828-80122a370af5.png)
![image](https://user-images.githubusercontent.com/72022193/162617881-19468b9a-47ff-448c-b002-701ec9f81cba.png)
![image](https://user-images.githubusercontent.com/72022193/162617894-3c95fc0a-1efb-44b3-962c-6137443b04c9.png)
Bytetrack:
![image](https://user-images.githubusercontent.com/72022193/162617938-47ae9815-3c06-4efe-8d58-a755f46d01c7.png)
![image](https://user-images.githubusercontent.com/72022193/162617930-bc56b229-23a0-4d73-ac30-0c64d24cf219.png)
![image](https://user-images.githubusercontent.com/72022193/162617949-5db3add4-bd7a-4788-b827-fe3b5ea96702.png)
then the indicators tested on val_half.json with the trained weights are the screenshots above.
I don't understand why the IDF1 metrics are so different (bytetrack-Reid: 64.5; bytetrack: 78.1).

**Comment by yanghaibin-cool** [comment]
@HanGuangXin Because the effect of yolox_s is not very good, I used yolox_x, the two screenshots above I used yolox_x to train for 80 rounds. I originally thought that I would get the indicators of the screenshots uploaded by your git repository, so I wanted to ask where the problem is, thank you very much.

**Comment by yanghaibin-cool** [comment]
> @yanghaibin-cool And these is something not right in `yolox_s_mot17_half.py`. The training should use `train_half.json`, not `train.json`.
> 
> And my results (mAP 0.556 MOTA 72.6) is from yolox_s, not yolox_x. The results of origin bytetrack is mAP 0.535 MOTA 71.5.
> 
> Please tell me which model is used (s or x).

Because the effect of yolox_s is not very good, I used yolox_x(yolox_x_mot17_half.py), the two screenshots above I used yolox_x to train for 80 rounds. I originally thought that I would get the indicators of the screenshots uploaded by your git repository, so I wanted to ask where the problem is, thank you very much.
Next, I am going to use yolox_s_mot17_half.py to train again.
Hope to get the same result as you.

**Comment by yanghaibin-cool** [comment]
> 

Maybe my computer configuration is too low（2080ti,11g video memory）

**Comment by yanghaibin-cool** [comment]
> And from my results, adding a reid head will **not** obviously increase forward time. There are several conflicts between your results and mine.
> 
> So, looking forward to your reply!

Maybe my computer configuration is too low（2080ti,11g video memory）

**Comment by yanghaibin-cool** [comment]
> For another thing, it is reasonable to see detection performance degradation after using an additional ReID head. You can go to CSTrack for the detailed reason.

OK, thanks for reminding.

**Comment by HanGuangXin** [comment]
@yanghaibin-cool Thanks for the reply.
There are some things:
1. My device is 2080Ti, too. But batchsize matters.
2. YOLOX_X with mAP 55.6 and MOTA 72.5 is not normal. YOLOX_S can get mAP 55.5, MOTA 71.8 and IDF1 73.8 (more or less) using `yolox_s_mot17_half.py`. So I think you can run yolox_s for comparison.
3. Indicators of the screenshots uploaded by myself in not right, which is obtained using `train.json`. So it is correct with 70+ MOTA. YOLOX_S with mAP 55.5, MOTA 71.8 and IDF1 73.8 is obtained using `train_half.json`, which should be correct.

Looking forward to more information!

**Comment by yanghaibin-cool** [comment]
> @yanghaibin-cool Thanks for the reply. There are some things:
> 
> 1. My device is 2080Ti, too. But batchsize matters.
> 2. YOLOX_X with mAP 55.6 and MOTA 72.5 is not normal. YOLOX_S can get mAP 55.5, MOTA 71.8 and IDF1 73.8 (more or less) using `yolox_s_mot17_half.py`. So I think you can run yolox_s for comparison.
> 3. Indicators of the screenshots uploaded by myself in not right, which is obtained using `train.json`. So it is correct with 70+ MOTA. YOLOX_S with mAP 55.5, MOTA 71.8 and IDF1 73.8 is obtained using `train_half.json`, which should be correct.
> 
> Looking forward to more information!

Thank you for your reply. Yes, I am training Bytetrack-Reid:
yolox_s_mot17_half.py(train_ann:train_half.json;val_ann:val_half).
Then train Bytetrack and compare the two.
Have you trained yolox_x_mot17_half, what is the result? (Could it be that Reid is not suitable for yolox_X_mot17_half?)

**Comment by HanGuangXin** [comment]
I will do it in one week or two. 
I'm a little busy now, sorry about that.

**Comment by yanghaibin-cool** [comment]
> I will do it in one week or two. I'm a little busy now, sorry about that.

May I ask what this function is for, it seems that there is no call in the code,
![f590d95d23be7ed54868d04b823777b](https://user-images.githubusercontent.com/72022193/162681367-ca16ae51-8480-494b-9f37-2c3f84b4454c.png)

**Comment by HanGuangXin** [comment]
just ignore it.



---Original---
From: "poplar ***@***.***&gt;
Date: Mon, Apr 11, 2022 14:55 PM
To: ***@***.***&gt;;
Cc: "Han ***@***.******@***.***&gt;;
Subject: Re: [HanGuangXin/ByteTrack_ReID] The tracked metric is too low (Issue#10)




  
I will do it in one week or two. I'm a little busy now, sorry about that.
  
May I ask what this function is for, it seems that there is no call in the code,

**Comment by yanghaibin-cool** [comment]
> just ignore it.
> […](#)
> ---Original--- From: "poplar ***@***.***&gt; Date: Mon, Apr 11, 2022 14:55 PM To: ***@***.***&gt;; Cc: "Han ***@***.******@***.***&gt;; Subject: Re: [HanGuangXin/ByteTrack_ReID] The tracked metric is too low (Issue#10) I will do it in one week or two. I'm a little busy now, sorry about that. May I ask what this function is for, it seems that there is no call in the code, — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***

**Comment by yanghaibin-cool** [comment]
> @yanghaibin-cool Thanks for the reply. There are some things:
> 
> 1. My device is 2080Ti, too. But batchsize matters.
> 2. YOLOX_X with mAP 55.6 and MOTA 72.5 is not normal. YOLOX_S can get mAP 55.5, MOTA 71.8 and IDF1 73.8 (more or less) using `yolox_s_mot17_half.py`. So I think you can run yolox_s for comparison.
> 3. Indicators of the screenshots uploaded by myself in not right, which is obtained using `train.json`. So it is correct with 70+ MOTA. YOLOX_S with mAP 55.5, MOTA 71.8 and IDF1 73.8 is obtained using `train_half.json`, which should be correct.
> 
> Looking forward to more information!

Hello! I have finished training with yolox_s_mot17_half(self.train_ann=train_half.json;self.val_ann=val_half.json).However,I didn't get the desired result(YOLOX_S with mAP 55.5, MOTA 71.8 and IDF1 73.8).
Below is the result of my training：
Bytetrack-Reid with train.py:
![image](https://user-images.githubusercontent.com/72022193/162733375-163d7732-bbd7-4307-883f-fa5529dde965.png)
Bytetrack-Reid with track.py:
COCO indicator：
![image](https://user-images.githubusercontent.com/72022193/162734324-23962d82-aaa8-480c-bfbb-8d3cc66463b9.png)
MOT indicator:
![image](https://user-images.githubusercontent.com/72022193/162734206-c156ddcd-cb75-4a07-a939-9542599ce19d.png)
Bytetrack with train.py:
![image](https://user-images.githubusercontent.com/72022193/162734728-93cff9ec-9fa7-42db-ab97-8ecceaf59597.png)
Bytetrack with track.py:
COCO indicator:
![image](https://user-images.githubusercontent.com/72022193/162734859-4164f186-3570-483a-9d10-216ff5bd8850.png)
MOT indicator:
![image](https://user-images.githubusercontent.com/72022193/162734934-cc6af28c-f5ff-4ac2-aeca-ac402814b771.png)
Why is the tracking effect worse after adding Reid? Where am I going wrong?
Looking forward to your reply
